Bash_scripts/HW28/Jenkinsfile

```diff
@@ -0,0 +1,48 @@
+pipeline {
+  agent any
+  stages {
+    stage('Shell script 0') {
+      steps {
+        sh '''#!/bin/bash
+
+cd $PATHTO
+SHELL_SCRIPTS=$(find . -name \'*.sh\')
+
+for var in $SHELL_SCRIPTS ; do
+shellcheck "$var"
+done'''
+      }
+    }    
+    stage('Shell script 1') {
+      steps {
+        sh '''#!/bin/bash
+
+DOCKERFILE=$(find "$PATHTO" -name \'Dockerfile\')
+
+for var in $DOCKERFILE ; do
+dockerfile_lint  -f "$var"
+done'''
+      }
+    }
+ stage('Push images') {
+  agent any
+  steps {
+    script {
+      def dockerImage = docker.build("${REPOUSR}/${NAMEIM}", "-f ${PATHTO}/Dockerfile .")
+      docker.withRegistry('', 'dockerhub') {
+        dockerImage.push("${VERSION}_${GIT_COMMIT}_${BUILD_NUMBER}")
+      }
+      echo "Pushed Docker Image: ${NAMEIM}"
+    }
+    sh "docker rmi ${REPOUSR}/${NAMEIM}:${VERSION}_${GIT_COMMIT}_${BUILD_NUMBER} "
+  }
+}   
+
+  }
+  parameters {
```

Comment:
Better to declare parameters in the beginning of your code.

Reply:
in all examples on the Internet, the parameters are indicated
 end of the code, and if you use the to Declaretive plugin, then too.

Reply:
Official Jenkins doc
https://www.jenkins.io/doc/book/pipeline/syntax/#available-parameters

Reply:
Putting all variables and parameters in the beginning  - best practise for almost everything.

Reply:
ok, thank you, fixed